src/xdsl/dialects/irdl.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from xdsl.dialects.builtin import ArrayAttr, StringAttr
+
+from xdsl.irdl import *
+from xdsl.ir import *
+
+
+@dataclass
+class IRDL:
+    ctx: MLContext
+
+    def __post_init__(self):
+        self.ctx.register_attr(EqTypeConstraintAttr)
+        self.ctx.register_attr(AnyTypeConstraintAttr)
+        self.ctx.register_attr(AnyOfTypeConstraintAttr)
+        self.ctx.register_attr(VarTypeConstraintAttr)
+        self.ctx.register_attr(DynTypeBaseConstraintAttr)
+        self.ctx.register_attr(DynTypeParamsConstraintAttr)
+        self.ctx.register_attr(TypeParamsConstraintAttr)
+        self.ctx.register_attr(NamedTypeConstraintAttr)
+
+        self.ctx.register_op(DialectOp)
+        self.ctx.register_op(ParametersOp)
+        self.ctx.register_op(TypeOp)
+        self.ctx.register_op(ConstraintVarsOp)
+        self.ctx.register_op(OperandsOp)
+        self.ctx.register_op(ResultsOp)
+        self.ctx.register_op(OperationOp)
+
+
+@irdl_attr_definition
+class EqTypeConstraintAttr(ParametrizedAttribute):
+
+    name = "equality_type_constraint"
+    data: ParameterDef[Attribute]
+
+    def enforce_one_param(data: Attribute) -> EqTypeConstraintAttr:
```

Comment:
I don't quite see why this is needed or what this is used for. Shouldn't this be a verifier?

Also notice that you are not returning an `EqTypeCostraintAttr` but rather an `Attribute`.